### PR TITLE
support shadow feature at IOS platform

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/platform/apple/modules/CanvasRenderingContext2DDelegate.mm [hunk 1]
```diff
@@ -348,6 +355,9 @@ - (void)fillText:(NSString *)text x:(CGFloat)x y:(CGFloat)y maxWidth:(CGFloat)ma
     CGContextSetShouldSubpixelQuantizeFonts(_context, false);
     CGContextBeginTransparencyLayerWithRect(_context, CGRectMake(0, 0, _width, _height), nullptr);
     CGContextSetTextDrawingMode(_context, kCGTextFill);
+    if (_shadowColor && (_shadowBlur > 0 || _shadowOffsetX > 0 || _shadowOffsetY > 0)) {
```
Comment (minggo) [inline native/cocos/platform/apple/modules/CanvasRenderingContext2DDelegate.mm]: How about add a function to check it, such as `isNeedSetShadow`. The function name should be clear.

**Review by minggo — APPROVED** [review]

File: native/cocos/platform/apple/modules/CanvasRenderingContext2DDelegate.mm [hunk 2]
```diff
@@ -413,6 +433,10 @@ - (void)setStrokeStyleWithRed:(CGFloat)r green:(CGFloat)g blue:(CGFloat)b alpha:
     _strokeStyle[3] = a;
 }
 
+- (void)setShadowColorWithRed:(CGFloat)r green:(CGFloat)g blue:(CGFloat)b alpha:(CGFloat)a {
+    _shadowColor = [NSColor colorWithRed:r green:g blue:b alpha:a];
```
Comment (dumganhar) [inline native/cocos/platform/apple/modules/CanvasRenderingContext2DDelegate.mm:440]: No need to retain it ?

File: native/cocos/platform/apple/modules/CanvasRenderingContext2DDelegate.mm [hunk 3]
```diff
@@ -643,4 +671,20 @@ - (void)lineToX:(float)x y:(float)y {
     }
 }
 
+void CanvasRenderingContext2DDelegate::setShadowBlur(float blur) {
+    _impl.shadowBlur = blur * 0.5f;
```
Comment (dumganhar) [inline native/cocos/platform/apple/modules/CanvasRenderingContext2DDelegate.mm:675]: Why need to `* 0.5f`?

Comment (bofeng-song) [reply native/cocos/platform/apple/modules/CanvasRenderingContext2DDelegate.mm:675]: same as android logic

**Review by dumganhar — APPROVED** [review]

**Pull request merged**